packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -562,39 +563,6 @@ export class ApolloServerBase {
 
     const extensions = [];
 
-    const schemaIsFederated = this.schemaIsFederated(schema);
-    const { engine } = this.config;
-    // Keep this extension second so it wraps everything, except error formatting
-    if (this.engineReportingAgent) {
-      if (schemaIsFederated) {
-        // XXX users can configure a federated Apollo Server to send metrics, but the
-        // Gateway should be responsible for that. It's possible that users are running
-        // their own gateway or running a federated service on its own. Nonetheless, in
-        // the likely case it was accidental, we warn users that they should only report
-        // metrics from the Gateway.
-        this.logger.warn(
-          "It looks like you're running a federated schema and you've configured your service " +
-            'to report metrics to Apollo Graph Manager. You should only configure your Apollo gateway ' +
-            'to report metrics to Apollo Graph Manager.',
-        );
-      }
-      extensions.push(() =>
-        this.engineReportingAgent!.newExtension(schemaHash),
-      );
```

Comment:
Only one ~extension~ plugin is created per server now, so we no longer have this factory function for initiation.